### PR TITLE
Add Google Test Unit tests

### PR DESCRIPTION
**Pull request by carolhmj**

### CODE REVIEW

File: Tests/UnitTests/Shared/Shared.cpp [hunk 1]
```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
...
-        {
-            Napi::Env env = info.Env();
-            exitCode.set_value(info[0].As<Napi::Number>().Int32Value());
-        }, "SetExitCode"));
+        env.Global().Set("SetExitCode", Napi::Function::New(
+                                            env, [&exitCode](const Napi::CallbackInfo& info) {
+                                                Napi::Env env = info.Env();
+                                                exitCode.set_value(info[0].As<Napi::Number>().Int32Value());
+                                            },
+                                            "SetExitCode"));
```
Comment (bghgary) [inline Tests/UnitTests/Shared/Shared.cpp:49]: nit: Is this formatting coming from clang-format? I thought our formatting doesn't do this indenting.

File: Tests/UnitTests/Shared/Shared.cpp [hunk 2]
```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
...
+        const char* test = messagesToTest[currIdx++];
+        if (strcmp(message, test))
+        {
+            fprintf(stdout, "Expected: %s, received: %s", test, message);
+            ADD_FAILURE();
+        }
+
+        fprintf(stdout, "[%s] %s", EnumToString(logLevel), message);
+        fflush(stdout);
+    })](Napi::Env env) mutable {
```
Comment (bghgary) [inline Tests/UnitTests/Shared/Shared.cpp:84]: Similar issue for the callback capture. No reason to capture if there is no state.

**Review by bghgary — APPROVED** [review]

File: Tests/UnitTests/Shared/Shared.cpp (hunk 1 above)
Comment (CedricGuillemet) [reply Tests/UnitTests/Shared/Shared.cpp:49]: fixed

File: Tests/UnitTests/Shared/Shared.cpp [hunk 3]
```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
...
     std::promise<int32_t> exitCode;
 
     Babylon::AppRuntime runtime{};
 
-    runtime.Dispatch([&exitCode, consoleCallback = std::move(consoleCallback)](Napi::Env env) mutable
-    {
+    runtime.Dispatch([&exitCode, consoleCallback = std::move([](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+        fprintf(stdout, "[%s] %s", EnumToString(logLevel), message);
+        fflush(stdout);
+    })](Napi::Env env) mutable {
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply Tests/UnitTests/Shared/Shared.cpp:36]: fixed

File: Tests/UnitTests/Shared/Shared.cpp (hunk 2 above)
Comment (CedricGuillemet) [reply Tests/UnitTests/Shared/Shared.cpp:84]: fixed

File: Tests/UnitTests/Shared/Shared.cpp [hunk 4]
```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
...
+{
+    std::promise<int32_t> exitCode;
+
+    Babylon::AppRuntime runtime{};
+
+    runtime.Dispatch([&exitCode, consoleCallback = std::move([](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+        static size_t currIdx = 0;
+        static std::vector<const char*> messagesToTest = {"foo bar\n"};
+        const char* test = messagesToTest[currIdx++];
+        if (strcmp(message, test))
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply Tests/UnitTests/Shared/Shared.cpp:76]: fixed

File: Tests/UnitTests/Shared/Shared.cpp [hunk 5]
```diff
@@ -23,36 +24,92 @@ const char* EnumToString(Babylon::Polyfills::Console::LogLevel logLevel)
...
+TEST(Console, ConsoleLog)
+{
+    std::promise<int32_t> exitCode;
+
+    Babylon::AppRuntime runtime{};
+
+    runtime.Dispatch([&exitCode, consoleCallback = std::move([](const char* message, Babylon::Polyfills::Console::LogLevel logLevel) {
+        static size_t currIdx = 0;
+        static std::vector<const char*> messagesToTest = {"foo bar\n"};
+        const char* test = messagesToTest[currIdx++];
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply Tests/UnitTests/Shared/Shared.cpp:75]: fixed

**Pull request merged**